### PR TITLE
Bedrock Renames

### LINKED ISSUE #1669: Bedrock API review
## Connections layer rename

Microsoft.AspNetCore.Http.Connections
Microsoft.AspNetCore.Http.Connections.Common
Microsoft.AspNetCore.Http.Connections.Client

```C#
app.UseConnections(routes =>
{
    // Handle SignalR chat connections on the /chat path (multiple transports)
    routes.MapConnection("chat", connectionBuilder =>
    {
        connectionBuilder.UseSignalR<Chat>();
    });
});
```

## Cleanup

- Rename DefaultConnectionContext to something else
- Remove Microsoft.AspNetCore.Sockets.Abstractions

We need to delete Sockets.Abstractions but `IConnection` is the last remaining hold out once we do the renames.

### PR DESCRIPTION
**Pull request by anurse**
Fixes #1669 

Lots of changes here. A summary:

* `Microsoft.AspNetCore.Sockets.Http` -> `Microsoft.AspNetCore.Http.Connections`
* `Microsoft.AspNetCore.Sockets.Client` -> `Microsoft.AspNetCore.Http.Connections.Client`
* `Microsoft.AspNetCore.Sockets.Common` -> `Microsoft.AspNetCore.Http.Connections.Common`
* `SocketsSample` -> `SignalRSamples`
* Make `ServerSentEventsParser` `internal` because a) I had to move it to the right folder path to fix the namespace and b) it's only used by tests

TODO:
* [x] Some final find-replace passes for "Sockets" (it's a little tricky because of WebSockets ;))
* [x] @davidfowl I can grab your Kestrel change tomorrow morning and yank Sockets.Abstractions. **FYI** Until this is done, it's OK to see `using Microsoft.AspNetCore.Sockets.Client` in the PR, since that's where `IConnection` resides. It's the only remaining thing named "Sockets"
* [x] Final review of Class View to make sure there aren't any strange namespaces hanging around.

### CODE REVIEW

**Comment by anurse** [comment]
Need a name for logging token `SocketsConnectionId`. cc @BrennanConroy @davidfowl

**Comment by davidfowl** [comment]
@anurse bro wrong base branch. This should be in release/2.1

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/ResponseUtils.cs [hunk 1]
```diff
@@ -6,9 +6,9 @@
 using System.Net;
 using System.Net.Http;
 using Microsoft.AspNetCore.Connections;
-using Microsoft.AspNetCore.Sockets;
 using Newtonsoft.Json;
-using SocketsTransportType = Microsoft.AspNetCore.Sockets.TransportType;
+
```
Comment (JamesNK) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/ResponseUtils.cs]: nit: blank line

File: samples/SignalRSamples/Hubs/Streaming.cs [hunk 2]
```diff
@@ -1,10 +1,10 @@
 using System;
```
Comment (JamesNK) [inline samples/SignalRSamples/Hubs/Streaming.cs]: What are the rules on copyright headers in samples? Some files have them and others don't.

**Comment by anurse** [comment]
Yeah, I'll try to update it before going to bed :). Autopilot...

File: samples/SignalRSamples/Hubs/Streaming.cs (hunk 2 above)
Comment (anurse) [reply samples/SignalRSamples/Hubs/Streaming.cs]: This seems like as good a time as any to get them consistent. They should all have headers. I'll review them

File: src/Microsoft.AspNetCore.Http.Connections.Client/ITransport.cs [hunk 3]
```diff
@@ -2,11 +2,12 @@
...
-using System.Threading.Tasks;
 using System.IO.Pipelines;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Sockets.Client;
 
-namespace Microsoft.AspNetCore.Sockets.Client
+namespace Microsoft.AspNetCore.Http.Connections.Client
 {
     public interface ITransport
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Connections.Client/ITransport.cs]: We can do another pass moving more things to pubternal.

File: clients/ts/FunctionalTests/FunctionalTests.csproj [hunk 4]
```diff
@@ -44,6 +45,7 @@
     <TypeScriptCompile Include="ts\ConnectionTests.ts" />
     <TypeScriptCompile Include="ts\HubConnectionTests.ts" />
     <TypeScriptCompile Include="ts\index.ts" />
+    <TypeScriptCompile Include="ts\TestLogger.ts" />
```
Comment (anurse) [inline clients/ts/FunctionalTests/FunctionalTests.csproj]: Looks like these ended up getting checked in. TBH I feel like we should leave them in as long as VS is going to keep insisting on creating them. Once VS is fixed, we can remove them.

File: src/Microsoft.AspNetCore.Http.Connections.Client/ITransport.cs (hunk 3 above)
Comment (anurse) [reply src/Microsoft.AspNetCore.Http.Connections.Client/ITransport.cs]: :+1:

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/ResponseUtils.cs (hunk 1 above)
Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/ResponseUtils.cs]: We haven't been terribly consistent about this but I was originally thinking that type alias `using`s should be separated from standard namespace-import `using`s.

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/ServerSentEventsParserTests.cs [hunk 5]
```diff
@@ -110,31 +110,31 @@ public void ParseSSEMessageIncompleteParseResult(string encodedMessage)
             var parser = new ServerSentEventsMessageParser();
             var pipe = new Pipe();
 
-                byte[] message = null;
-                SequencePosition consumed = default, examined = default;
+            byte[] message = null;
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/ServerSentEventsParserTests.cs]: I Ctrl-K+Ctrl-D'd most of the files I cleaned up and things like this are what happened as a result :)

Comment (davidfowl) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/ServerSentEventsParserTests.cs]: You should do that when you code normally.

File: test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs [hunk 6]
```diff
@@ -29,7 +28,9 @@
 using Xunit;
 using Xunit.Abstractions;
 
-namespace Microsoft.AspNetCore.Sockets.Tests
+// Needed to diambiguate with System.Net.TransportType
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs]: ?

Comment (anurse) [reply test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs]: You mean that didn't disambiguate you?

**Comment by anurse** [comment]
Interesting:

```
        ....
        - level: Error
          message: Server returned handshake error: An unexpected error occurred during connection handshake. InvalidDataException: Missing required property 'protocol'.
        - level: Trace
          message: (LongPolling transport) polling: http://127.0.0.1:54817/testhub?id=Rs8ARL1N29TegQvMWMF4SA&_=1522428941594
        - level: Error
          message: Connection disconnected with error 'Error: Server returned handshake error: An unexpected error occurred during connection handshake. InvalidDataException: Missing required property 'protocol'.'.
```

**Comment by davidfowl** [comment]
Oh no! It showed up again 😨

**Comment by anurse** [comment]
Sockets.Abstractions is removed! This is ready to go once approved!

File: src/Microsoft.AspNetCore.SignalR.Common/Microsoft.AspNetCore.SignalR.Common.csproj [hunk 7]
```diff
@@ -16,11 +16,8 @@
     <PackageReference Include="System.Memory" Version="$(SystemMemoryPackageVersion)" />
     <PackageReference Include="System.Buffers" Version="$(SystemBuffersPackageVersion)" />
     <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafePackageVersion)" />
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR.Common/Microsoft.AspNetCore.SignalR.Common.csproj]: We need to clean up this dependency graph (took a mental note)

File: test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj [hunk 8]
```diff
@@ -33,6 +33,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Connections.Abstractions" Version="$(MicrosoftAspNetCoreConnectionsAbstractionsPackageVersion)" />
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: I don't think you need this.

Comment (anurse) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: I'll double check

File: src/Microsoft.AspNetCore.SignalR.Common/Microsoft.AspNetCore.SignalR.Common.csproj (hunk 7 above)
Comment (anurse) [reply src/Microsoft.AspNetCore.SignalR.Common/Microsoft.AspNetCore.SignalR.Common.csproj]: > ~~took a mental note~~ forgot about it immediately

FTFY

**Comment by davidfowl** [comment]
Build failures...

**Comment by JamesNK** [comment]
`The active test run was aborted. Reason: Unhandled Exception: Unhandled Exception: System.InvalidOperationException: Another continuation was already registered.`

AppVeyor failure is caused by that corefx sockets bug I think.

**Comment by JamesNK** [comment]
Merge conflicts tho...

**Comment by anurse** [comment]
:up: :date: merge conflicts resolved

**Comment by anurse** [comment]
Verbal approval from @davidfowl

**Pull request merged**